cpu/atmega_common/periph/gpio.c

```diff
@@ -154,35 +164,27 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     /* clear global interrupt flag */
     cli();
 
-#if defined(CPU_ATMEGA328P)
-    /* INT pins start at PD2 instead of at PD0 */
-    pin_num -= 2;
-#elif defined(CPU_ATMEGA1284P)
-    /* INT pins start at PD2 instead of at PD0 on PORT_D */
-    if (_port_num(pin) == PORT_D) {
-        pin_num -= 2;
-    }
-#endif
-
-    EIMSK |= (1 << pin_num);
+    /* enable interrupt number int_num */
+    EIMSK |= (1 << int_num);
 
     /* configure the flank */
     if (flank > GPIO_RISING) {
         return -1;
     }
 
-    if (pin_num < 4) {
-        EICRA |= (flank << (pin_num * 2));
+    /* apply flank to interrupt number int_num */
+    if (int_num < 4) {
+        EICRA |= (flank << (int_num * 2));
     }
 #if defined(EICRB)
     else {
-        EICRB |= (flank << (pin_num * 2) % 4);
+        EICRB |= (flank << ((int_num % 4) * 2));
```

Comment:
Can you tell more about this change? At a first glance it looks like the same, but I know there's a difference. I also wonder it worked in the past.

Reply:
Here is an example if int_num == 6:
- Old behavior: multiply by 2 (== 12), then modulo 4 (remainder of 12/4 = 0), ergo use ISC40 and ISC41 in EICRB (wrong).
- New behavior: modulo 4 (remainder of 6/4 = 2), then multiply by 2 (2*2 = 4), ergo use ISC60 and ISC61 in EICRB (right).

Reply:
Ok, thanks for the explanation!